src/ibex_bluesky_core/plans/reflectometry/autoalign_utils.py

```diff
@@ -0,0 +1,232 @@
+"""A general tool for reflectometers to use to save time aligning their beamlines."""
+
+from collections.abc import Generator
+from typing import Callable
+
+import bluesky.plan_stubs as bps
+from bluesky.protocols import NamedMovable
+from bluesky.utils import Msg
+from lmfit.model import ModelResult
+from typing_extensions import NotRequired, TypedDict, Unpack
+
+from ibex_bluesky_core.callbacks import ISISCallbacks
+from ibex_bluesky_core.callbacks.fitting import FitMethod
+from ibex_bluesky_core.devices.simpledae import SimpleDae
+from ibex_bluesky_core.plan_stubs import call_sync
+from ibex_bluesky_core.plans import scan
+
+
+def _check_parameter(
+    alignment_param_value: float,
+    result: ModelResult,
+    init_mot_pos: float,
+    rel_scan_range: float,
+    is_good_fit: Callable[[ModelResult, float], bool] | None = None,
+) -> bool:
+    """Check that the optimised value is within scan range and then runs user provided checks.
+
+    Returns True means the found result was sensible.
+
+    Args:
+        alignment_param_value (float): The name of the optimised value.
+        result (ModelResult): The fitting resuls returned from lmfit.
+        init_mot_pos (float): The initial motor position before scanning.
+        rel_scan_range (float): The current relative scan range.
+        is_good_fit (Callable[[ModelResult, float], bool] | None): User provided checks on the
+            optimised value, must return True if result is sensible.
+
+    Returns:
+        True if value is sensible. False otherwise.
+
+    """
+    if init_mot_pos + rel_scan_range < alignment_param_value:
```

Comment:
may want to raise a `ValueError` here with a useful message that is caught outside - otherwise a user might be "in the dark" as to what has failed

Reply:
maybe make a new exception for this